src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java

```diff
@@ -363,4 +363,25 @@ public static boolean isInScope(DetailAST ast, Scope scope) {
         return surroundingScopeOfAstToken == scope;
     }
 
+    /**
+     * Check if given parameter definition is for a static, default or private method.
+     *
+     * @param parameterDefAst parameter definition
+     * @return true if it is for a static, default or private method, false otherwise
+     */
+    public static boolean isStaticDefaultOrPrivateMethod(DetailAST parameterDefAst) {
```

Comment:
plaase  move method to Check, it is very specific methods and unlikely will be reused, so better let it serv a goal of Check not a general.

CI detected bunch of not covered code by tests.
https://github.com/checkstyle/checkstyle/actions/runs/12841792273/job/35819225523?pr=16132#step:7:28
it should less ones method is moved to Check and all tests are run.

but it there is any still be missed.
please try to cover code with extra testing, let me know if you need more details on what is pitest and how to deal with it.

Reply:
If I move it to check, it fails another checkstyle cause there are then too many methods in FinalLocalVariableCheck

Reply:
We can suppress it at https://github.com/checkstyle/checkstyle/blob/master/config/suppressions.xml